Full page alert banner

Adnan-cds opened the pull request:
*Full page* bundle for the Alert banner entity.

Contains config files, theme templates, and a related asset library.

This feature is totally dependent on Bootstrap making it unsuitable for general use outside the localgov_theme.  The next step is to replace the Bootstrap specific code with plain Javascript and CSS.

Closes #134 

FAO @andybroomfield @cjstevens78 @tom-steel @willguv

Adnan-cds committed:
Full page alert banner.

Custom theme template and supporting asset library.

Adnan-cds committed:
Full page alert banner.

Config files for the *Full page* alert banner bundle.

Adnan-cds committed:
Full page alert banner.

Removed UUIDs from config files.

Adnan-cds committed:
Full page alert banner.

Updating module dependencies.

Adnan-cds committed:
Full page alert banner proposal.

Fixed Composer dependency for localgov_core.

Adnan-cds commented on composer.json:
```diff
@@ -5,6 +5,7 @@
     "require": {
-        "drupal/field_group": "~3.1"
+        "drupal/field_group": "~3.1",
+        "localgovdrupal/localgov_media": "^1.0@alpha"
```
Thanks Andy.

Adnan-cds replied:
localgovdrupal/localgov_core has now moved into the new ```suggests``` section as per Andy's suggestion.

Adnan-cds commented:
Hi Andy,
A little clarification.  This change request is more of a demo than a merge request.  You may remember I mentioned in Slack that I will pass you the config files we have used for the Full page alert banner.  So here they are.  I don't have any preference for keeping the Full page banner functionality within the localgov_alert_banner module or in one of its submodule or somewhere else.

Adnan-cds committed:
Revert "Full page alert banner proposal."

This reverts commit 00bea9a313fdcb3a64d6ad025e677cac28f1922e.

Adnan-cds committed:
Revert "Full page alert banner."

This reverts commit c82aeaba38c19a7d7cbdcd14b509b9215dbc8208.

Adnan-cds committed:
Revert "Full page alert banner."

This reverts commit 536830c5e7165019ca30a41e17af0d2c919e09cd.

Adnan-cds committed:
Revert "Full page alert banner."

This reverts commit 05795be1ea6246e8278e1d6d7353d6804bd3e8e9.

Adnan-cds committed:
Revert "Full page alert banner."

This reverts commit 451fcc5386ee095087a60ea74e114736620bc966.

Adnan-cds committed:
Full page alert banner: Adding submodule dependency.

Adnan-cds committed:
Full page alert banner submodule.

Adnan-cds committed:
Full page alert banner: Fixing version constraints.

Adnan-cds committed:
Full page alert: Coding standard fixes.

Adnan-cds committed:
Merge remote-tracking branch 'origin/1.x' into feature/full-page-alert

Adnan-cds commented:
Hi Andy,
I have turned the Full page alert banner code into a submodule.  Chris is working to replace the Bootstrap-specific parts with plain Javascript and CSS.  Once that's done, we hope to take this change from Draft to Ready-for-review.

@andybroomfield @cjstevens78

andybroomfield commented:
Thanks @Adnan-cds 
@willguv @finnlewis has this change been approved by product group / working group? Following on from the slack discussions.

Adnan-cds committed:
Full page alert: More style rules.

cjstevens78 committed:
styling fixes for full page alert banner

Adnan-cds committed:
Full page alert: Fixed CSS property.

Replaced "$carbon" with corresponding colour code.

Adnan-cds committed:
Merge remote-tracking branch 'origin/1.x' into feature/full-page-alert

Adnan-cds committed:
Full page alert banner: Link styling.

Text colour for link text within the modal body content.

Adnan-cds committed:
Full page alert banner: Visibility field config.

Commit:
Merge remote-tracking branch 'origin/1.x' into feature/full-page-alert

cjstevens78 committed:
remove bootstrap and jquery dependencies

cjstevens78 committed:
merging with lgd releaseMerge branch 'feature/full-page-alert' of https://github.com/localgovdrupal/localgov_alert_banner into feature/full-page-alert

Adnan-cds commented:
Hi Chris,
I have tried to use this on a vanilla LGD site.  It is throwing a Javascript error saying ```favDialog.showModal is not a function```.  I cannot see any code providing ```favDialog```.  My guess is, perhaps a file providing favDialog is not in this repo.  Could you please double check.

@cjstevens78

cjstevens78 committed:
remove dialog css not required

cjstevens78 committed:
tidy up js fix various ff issues and rename the modal id to be more appropriate

cjstevens78 commented:
@Adnan-cds ive re-worked this - had a tidy up and fixed the firefox problems.

Adnan-cds committed:
Full page alert banner.

Doing away with fixed DOM Id values.

Reverting from localStorage to Cookies for storing Alert banner state.

Adnan-cds committed:
Merge remote-tracking branch 'origin/1.x' into feature/full-page-alert

Adnan-cds committed:
Merge remote-tracking branch 'origin/1.x' into feature/full-page-alert

Adnan-cds committed:
Merge branch 'feature/full-page-alert' of https://github.com/localgovdrupal/localgov_alert_banner into feature/full-page-alert

cjstevens78 committed:
ff vertical centering fix

Adnan-cds committed:
Full page alert.

localgov_core is now a suggested dependency.  It is only required when the
localgov_alert_banner_full_page *submodule* is enabled.

Adnan-cds committed:
Merge remote-tracking branch 'origin/1.x' into feature/full-page-alert

Adnan-cds commented:
Ready for review.  Finally :)

To summarize the change, it adds a **submodule** called localgov_alert_banner_full_page that can be enabled or disabled as needed.  When enabled, it offers a bundle (i.e. variant) of the standard Alert banner that can be edited like any other Alert banner.

@andybroomfield @danchamp  @finnlewis  @tom-steel @stephen-cox @willguv

andybroomfield reviewed (commented):
My glance through at the code says this is fine.
Is there any other council that implements (or is planning on implementing) a full screen alert banner able to test and review this as I'm going to be away.
@danchamp @finnlewis (Lambeth?) or others.

finnlewis committed:
Fix merge conflict with 1.x branch.

finnlewis commented:
@Adnan-cds I merged back from the 1.x branch to bring in the latest changes and resolve merge conflict in composer.json.

I am now testing locally and when creating a full page alert, the moderation state form element is displayed on the "Are you sure you want to set live?' page. This seems to prevent me clicking the confirm, so I think the forms are conflicting in some way.

Have you seen this? 

I assume the desired behaviour is to NOT see the moderation state, in line with the default alert banner variant. 

See screenshot below.

![image](https://user-images.githubusercontent.com/326588/132819222-689db6d8-929d-4362-afdd-85f50290dff3.png)

finnlewis committed:
Hide the moderation control from the entity view display for the full page alert banner.

finnlewis commented:
@Adnan-cds I fixed that issue with https://github.com/localgovdrupal/localgov_alert_banner/pull/133/commits/941536aef5f12a33c9596a9416a6329c65a36c89

finnlewis commented:
When logged in, the full page alert is not full page, which is a bit jarring from a UX perspective.

![image](https://user-images.githubusercontent.com/326588/132822298-d9e8b893-c0b6-4301-af5d-d0d999fae490.png)

I'm guessing with a bit of CSS tinkering we can make it behave the same logged in or logged out.

@Adnan-cds / @cjstevens78  it would be good to get this resolved this sprint, do you have time to look at this today? 

And perhaps @markconroy could you take a quick look at the css / js in this pull request?

Adnan-cds commented:
> I fixed that issue with 941536a

Thanks Finn :)

> it would be good to get this resolved this sprint, do you have time to look at this today?

I will leave it for Chris.

I like the "Big bad news" image :D

finnlewis commented:
@cjstevens78 are you able to take a look at this? It would be good to get this merged today or tomorrow if at all possible.... to coincide with the end of sprint. 

If not, if you can point me in the right direction....  :)

cjstevens78 commented:
> @cjstevens78 are you able to take a look at this? It would be good to get this merged today or tomorrow if at all possible.... to coincide with the end of sprint.
> 
> If not, if you can point me in the right direction.... :)

@finnlewis I cant replicate this locally - although thinking about it I wonder if its maybe a good thing for it not to completely hijack the page when logged in because the user in this instance wants to see the site they are interacting with and aren't required to focus their attention on the pop up.

cjstevens78 commented:
@finnlewis we've just had a discussion about this at standup - could we get this merged in as is and look at the logged in overlay as a separate ticket?

finnlewis commented:
Sure thing, happy to merge as is.

finnlewis reviewed (approved):
Looks good to me!

Adnan-cds commented:
Thanks Finn.  Merging...
